src/avram/schema_enforcer/validation.cr

```diff
@@ -1,14 +1,20 @@
 abstract class Avram::SchemaEnforcer::Validation
   private getter model_class : Avram::Model.class
-  private getter database_info : Avram::Database::DatabaseInfo
 
   def initialize(@model_class)
-    @database_info = @model_class.database.database_info
```

Comment:
This could no longer live here because `#database_info` causes a database query which must happen after the database has been configured. Moving it into the class level of models meant that it is no longer the case.